### PR TITLE
fix: locking on linux and osx

### LINKED ISSUE #1: flock error: bad number
All attempts to use kvstore result in the error 

    flock: bad number: '/home/Matthew/.kvstore/testfile.lock'

A little digging around via `man flock` reveals that the call 

    flock -x -w 5 "$lockfile"

isn't going to work.  You have to call `flock` with some kind of command OR replace the `$lockfile` with a file descriptor number. This is from the man page for `flock` on two different Linux systems. Here is a copy from the net:

[http://www.tutorialspoint.com/unix_commands/flock.htm](url)

Please advise.

### PR DESCRIPTION
**Pull request by ccarpita**
Should fix https://github.com/ccarpita/kvstore/issues/1

Also, fix a number of shell style warnings.

### CODE REVIEW

**Comment by ccarpita** [comment]
@matthewpersico if you get a chance, can you test this in your environment and let me know if this addresses the issue?

File: kvstore.sh [hunk 1]
```diff
@@ -69,14 +56,25 @@ _lock_then () {
...
-  if ! _lockfile "$lockfile"; then
-    echo "error: could not aquire lock" &>2
-    return 2
+  if type flock &>/dev/null; then
+    set -e
+    (
+      flock -w 5 -x 200
+      "$cmd" "$@"
+    ) 200>"$lockfile"
+    set +e
```
Comment (matthewpersico) [inline kvstore.sh:65]: LOL! You must have read the exact same docs I did, because my fixes ALSO use 200 for the fd. :-)

**Comment by matthewpersico** [comment]
I'll try to pull this down this weekend and review.

File: kvstore.sh (hunk 1 above)
Comment (matthewpersico) [reply kvstore.sh:65]: Oh wait - I gave you the link to what I read in the issue. Duh!

Comment (ccarpita) [reply kvstore.sh:65]: I actually hadn't followed the link, but I had read that same doc earlier this week for an unrelated problem 😆

**Comment by matthewpersico** [comment]
Approved! Works fine. On Cygwin, no less. :-)

**Comment by matthewpersico** [comment]
If you merge back into master, I can then send you a PR with my other changes.

**Pull request merged**